crates/pop-cli/src/commands/call/parachain.rs

```diff
@@ -0,0 +1,809 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::cli::{self, traits::*};
+use anyhow::{anyhow, Result};
+use clap::Args;
+use pop_parachains::{
+	construct_extrinsic, encode_call_data, find_extrinsic_by_name, find_pallet_by_name,
+	parse_chain_metadata, set_up_client, sign_and_submit_extrinsic, supported_actions, Action,
+	DynamicPayload, Extrinsic, OnlineClient, Pallet, Param, SubstrateConfig,
+};
+use url::Url;
+
+const DEFAULT_URL: &str = "ws://localhost:9944/";
+const DEFAULT_URI: &str = "//Alice";
+
+/// Command to execute extrinsics with configurable pallets, arguments, and signing options.
+#[derive(Args, Clone)]
+pub struct CallParachainCommand {
+	/// The pallet containing the extrinsic to execute.
+	#[arg(short, long, value_parser = parse_pallet_name)]
+	pallet: Option<String>,
+	/// The extrinsic to execute within the chosen pallet.
+	#[arg(short, long, value_parser = parse_extrinsic_name)]
+	extrinsic: Option<String>,
+	/// The extrinsic arguments, encoded as strings.
+	#[arg(short, long, num_args = 0..,)]
+	args: Vec<String>,
+	/// Websocket endpoint of a node.
+	#[arg(short, long, value_parser)]
+	url: Option<Url>,
+	/// Secret key URI for the account signing the extrinsic.
+	///
+	/// e.g.
+	/// - for a dev account "//Alice"
+	/// - with a password "//Alice///SECRET_PASSWORD"
+	#[arg(short, long)]
+	suri: Option<String>,
+	/// Automatically signs and submits the extrinsic without prompting for confirmation.
+	#[arg(short('y'), long)]
+	skip_confirm: bool,
+}
+
+impl CallParachainCommand {
+	/// Executes the command.
+	pub(crate) async fn execute(mut self) -> Result<()> {
+		let mut cli = cli::Cli;
+		// Check if all fields are specified via command line argument.
+		let prompt_to_repeat_call = self.requires_user_input();
+		// Configure the chain.
+		let chain = self.configure_chain(&mut cli).await?;
+		loop {
+			// Configure the call based on command line arguments/call UI.
+			let mut call = match self.configure_call(&chain, &mut cli).await {
+				Ok(call) => call,
+				Err(e) => {
+					display_message(&e.to_string(), false, &mut cli)?;
+					break;
+				},
+			};
+			// Display the configured call.
+			cli.info(call.display(&chain))?;
+			// Prepare the extrinsic.
+			let tx = match call.prepare_extrinsic(&chain.client, &mut cli).await {
+				Ok(payload) => payload,
+				Err(e) => {
+					display_message(&e.to_string(), false, &mut cli)?;
+					break;
```

Comment:
I've noticed that when dispatching a system remark call, if the parameter is not prefixed by `0x` this causes an error.

I was wondering if it would be beneficial prompting again in case the user wants to repeat the call and amend the possible errors.
Or maybe this would just include an extra unnecessary interaction.

Reply:
Yes, it seems the panic is caused by `subxt` when preparing the call with incorrect arguments: https://github.com/paritytech/subxt/blob/master/core/src/tx/payload.rs#L186. 

In PolkadotJS it does not require the 0x prefix and automatically encodes the value into a hexadecimal format. This approach seems more user-friendly and would be a good model to follow.
In the existing PR https://github.com/r0gue-io/pop-cli/pull/363 we are specifically handling  (`TypeDef::Sequence`) types so I think is better to add the fix there: https://github.com/r0gue-io/pop-cli/pull/363/commits/3f79680b2c97ac3a30192f80961052b254c042f5 to ensure our logic aligns with PolkadotJS.

Might be easiest approach what you are suggesting. I leave the comment open.

Another idea was to wrap the `panic! ` coming from subxt, but this probably adds unnecessary complexity.

Reply:
Makes sense! Marking as resolved.